File: Core/Gauge/CS/GaugeDemo/PageInteractive.cs
using System;
using System.Collections.Generic;
using System.ComponentModel;
using System.Data;
using System.Drawing;
using System.Text;
using System.Windows.Forms;
using C1.Win.Gauge;

namespace GaugeDemo
{
    public partial class PageInteractive : BaseGaugePage
    {
        public PageInteractive()
        {
            InitializeComponent();
        }

        private void c1RadialGauge1_ItemMouseMove(object sender, ItemMouseEventArgs e)
        {
            if ((e.Button & MouseButtons.Left) == MouseButtons.Left)
            {
                C1GaugeEllipse el = e.Item as C1GaugeEllipse;
                if (el != null && el.Name == "back")
                {
                    C1GaugePointer p = e.Gauge.Pointer;
                    p.UpdateValue(p.GetValueAt(e.X, e.Y), 2.5);
                }
            }
        }

        private void c1RadialGauge1_FormatLabel(object sender, FormatLabelEventArgs e)
        {
            if (double.IsNaN(e.Value))
            {
                e.Text = "Off";
            }
        }
    }
}

File: NetFramework/Gauge/VB/GaugeDemo/PageInteractive.vb
Imports C1.Win.Gauge

Public Class PageInteractive

    Private Sub c1RadialGauge1_ItemMouseMove(ByVal sender As System.Object, ByVal e As ItemMouseEventArgs) Handles C1RadialGauge1.ItemMouseMove, C1RadialGauge1.ItemMouseDown
        If TypeOf e.Item Is C1GaugeEllipse AndAlso (e.Button And MouseButtons.Left) = MouseButtons.Left Then
            Dim el As C1GaugeEllipse = CType(e.Item, C1GaugeEllipse)
            If el.Name = "back" Then
                Dim p As C1GaugePointer = e.Gauge.Pointer
                p.UpdateValue(p.GetValueAt(e.X, e.Y), 2.5)
            End If
        End If
    End Sub

    Private Sub c1RadialGauge1_FormatLabel(ByVal sender As System.Object, ByVal e As FormatLabelEventArgs) Handles C1RadialGauge1.FormatLabel
        If Double.IsNaN(e.Value) Then
            e.Text = "Off"
        End If
    End Sub

End Class
